hkl/diffract.py

```diff
@@ -112,65 +140,91 @@ def __init__(self, prefix, calc_kw=None, decision_fcn=None,
+            configuration_attrs=configuration_attrs,
+            **kwargs
+        )
 
         if read_attrs is None:
             # if unspecified, set the read attrs to the pseudo/real motor
             # positions once known
-            self.read_attrs = (list(self.PseudoPosition._fields) +
-                               list(self.RealPosition._fields))
+            self.read_attrs = list(self.PseudoPosition._fields) + list(
+                self.RealPosition._fields
+            )
 
-        self.energy.subscribe(self._energy_changed,
-                              event_type=Signal.SUB_VALUE)
+        self.energy.subscribe(
+            self._energy_changed, event_type=Signal.SUB_VALUE)
 
     def _energy_changed(self, value=None, **kwargs):
-        '''
+        """
         Callback indicating that the energy signal was updated
-        '''
-        logger.debug('{0.name} energy changed: {1}'.format(self, value))
+        """
+        logger.debug(f"{self.name} energy changed: {value}")
         self._calc.energy = value
         self._update_position()
 
     @property
     def calc(self):
-        '''The calculation instance'''
+        """The calculation instance"""
         return self._calc
 
     @property
     def engine(self):
-        '''The calculation engine associated with the diffractometer'''
+        """The calculation engine associated with the diffractometer"""
         return self._calc.engine
 
     # TODO so these calculations change the internal state of the hkl
-    # calculation class, which is probably not a good thing -- it becomes a
-    # problem when someone uses these functions outside of move()
+    # calculation class, which is probably not a good thing
+    # -- it becomes a problem when someone uses these functions
+    # outside of move()
 
     @pseudo_position_argument
     def forward(self, pseudo):
-        solutions = self._calc.forward_iter(start=self.position, end=pseudo,
-                                            max_iters=100)
-        logger.debug('pseudo to real: {}'.format(solutions))
+        solutions = self._calc.forward_iter(
+            start=self.position, end=pseudo, max_iters=100
+        )
+        logger.debug(f"pseudo to real: {solutions}")
         return self._decision_fcn(pseudo, solutions)
 
     @real_position_argument
     def inverse(self, real):
         self._calc.physical_positions = real
         return self.PseudoPosition(*self._calc.pseudo_positions)
 
+    def check_value(self, pos):
+        """
+        Raise exception if pos is not within limits.
+
+        In a scan, a subset of the pseudo axes may be directed,
+        which are given in a dict from a set message from the
+        bluesky RunEngine.
+
+        It is not permitted to scan both pseudo and real positioners.
+        """
+        if isinstance(pos, dict):
+            # Redefine and fill in any missing values.
+            pos = [
+                pos.get(p.attr_name) or p.position
```

Comment:
```suggestion
                pos.get(p.attr_name, p.position)
```

Using the `or` short circuit would replace a user supplied value of 0 with the current position!

Reply:
Yikes!  I'll code differently then.

Reply:
Done.  Next time, I'll push the *Commit suggestion* button on GitHub.